lib/modules/string/replace.dart

```diff
@@ -0,0 +1,4 @@
+String replace(String string, dynamic pattern, dynamic replacement) {
+  Pattern patternObj = pattern is String ? RegExp(pattern) : pattern;
```

Comment:
This condition might be safer if we evaluate it the other way around as follows:-

```dart
Pattern patternObj = pattern is RegExp ? pattern : RegExp(pattern.toString());
```